mlkem/rej_uniform_asm.s

```diff
@@ -0,0 +1,268 @@
+// SPDX-License-Identifier: Apache-2.0
+#include "params.h"
+
+.text:
+
+.macro div48    dst, src
+    mov     w8, #43691
+    movk    w8, #43690, lsl #16
+    umull   x8, \src, w8
+    lsr     x8, x8, #37
+    add     w8, w8, w8, lsl #1
+    lsl     \dst, w8, #4
+.endm
+/*************************************************
+* Name:        rej_uniform_asm
+*
+* Description: Full SIMD lane, run rejection sampling on uniform random bytes to generate
+*              uniform random integers mod q
+*
+* Arguments:   - int16_t *r:          pointer to output buffer
+*              - unsigned int len:    requested number of 16-bit integers
+*                                     (uniform mod q)
+*              - const uint8_t *buf:  pointer to input buffer
+*                                     (assumed to be uniform random bytes)
+*              - unsigned int buflen: length of input buffer in bytes
+*              - unsigned int *consumed: the length of consumed buffer
+* Returns number of sampled 16-bit integers (at most len)
+**************************************************/
+.align 4
+.global     rej_uniform_asm
+.global     _rej_uniform_asm
+rej_uniform_asm:
```

Comment:
Don't you need some preamble/postamble for stack stash/restore?

Reply:
So I check, register from x8 to x17 can be used freely by callee as temporary register. Register x19-x28 must be reserved, I tried not to use x19-x28 so I think the current approach has no problem.

Reply:
You might need some more registers when scheduling the code through SLOTHY, but we can add the necessary prologue/epilogue then.

For now, can you add a comment / warning please?

Reply:
Comment added in https://github.com/pq-code-package/mlkem-c-aarch64/pull/47/commits/f4ead9afa2b892eed146a027a4308c97e4771b9b. 
Thanks for the suggestion @hanno-becker !
I think at kernel level (if this work is adopted to kernel), people would use intra-procedure-call scratch registers `x16, x17`, thus I stick to strict ARM64 calling convention. `x8, x16, x17, x29, x30` registers are preserved.